src/styles/style.scss

```diff
@@ -0,0 +1,70 @@
+$cell-size: 100px;
+$daysGap: 1px;
+$daysCount: 31;
+$calendarPaddings: 10px;
+$week-days: (mon 0, tue 1, wed 2, thu 3, fri 4, sun 5, sat 6);
+$width: 7 * $cell-size + 6 * $daysGap;
+
+body {
+  margin: 0;
+  height: 100vh;
+  box-sizing: border-box;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  width: $width;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $daysGap;
+
+  &__day {
+    box-sizing: border-box;
+    background-color: #eee;
+    width: 100px;
+    height: 100px;
+    align-content: center;
+    align-items: center;
+    text-align: center;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    border: 1px solid black;
+    cursor: pointer;
+  }
+
+  @for $countCalendarDay from 1 through $daysCount {
+    &__day:nth-child(#{$countCalendarDay})::before {
+      content: '#{$countCalendarDay}';
+    }
+  }
+
+  // @for $lengthCalendarDay from 28 through 31 {
+  //   &__day:nth-child(#{$lengthCalendarDay}) {
+  //     display: none;
+  //   }
+  // }
+
+  @for $lengthCalendarDay from 28 through 31 {
+    &--month-length-#{$lengthCalendarDay}
+      :nth-child(n + #{$lengthCalendarDay + 1}) {
+      display: none;
+    }
+  }
+
+  // Margin left and start day in calendar
+  @each $day, $numDay in $week-days {
+    &--start-day-#{$day} > &__day:first-child {
+      margin-left: ($cell-size + $daysGap) * $numDay;
+    }
+  }
+
+  @for $transformationCalendar from 1 through $daysCount {
+    &__day:nth-child(#{$transformationCalendar}):hover {
+      transform: translateY(-20px);
+      transition: 500ms;
+      background-color: #ffbfcb;
```

Comment:
Consider using these styles in the regular class instead of generating the same style for 31 elements.